hub/log.go

```diff
@@ -1,44 +1,24 @@
 package hub
 
-import (
-	fluentd "github.com/joonix/log"
-	log "github.com/sirupsen/logrus"
-	"github.com/spf13/viper"
-)
+import "go.uber.org/zap/zapcore"
 
-func addUpdateFields(f log.Fields, u *Update, debug bool) log.Fields {
-	f["event_id"] = u.ID
-	f["event_type"] = u.Type
-	f["event_retry"] = u.Retry
-	f["update_topics"] = u.Topics
-	f["update_private"] = u.Private
+type stringArray []string
 
-	if debug {
-		f["update_data"] = u.Data
+func (ss stringArray) MarshalLogArray(arr zapcore.ArrayEncoder) error {
+	for i := range ss {
+		arr.AppendString(ss[i])
 	}
 
-	return f
+	return nil
 }
 
-func createFields(u *Update, s *Subscriber) log.Fields {
-	f := addUpdateFields(log.Fields{}, u, s.Debug)
-	for k, v := range s.LogFields {
-		f[k] = v
-	}
-
-	return f
-}
+type Field = zapcore.Field
```

Comment:
This is an unfortunate consequence of Go's type system. Maybe add a comment here saying that generics will allow you to define your own contract for the `Field` interface as well.